### PR TITLE
Allow reindexing only recent records.

### PR DESCRIPTION
**Pull request by olmar**
Implement ``get_updated_field`` in order to use ``update_index`` command with the start and end dates for indexing within.
Issue: https://openedx.atlassian.net/browse/TNL-1106

More info:
* [SearchIndex.get_updated_field(self)](http://django-haystack.readthedocs.org/en/latest/searchindex_api.html?highlight=get_updated_field#SearchIndex.get_updated_field)
* [haystack ``update_index`` command](http://django-haystack.readthedocs.org/en/latest/management_commands.html#update-index)

Please review: @tymofij 
FYI: @e0d @jimabramson

### CODE REVIEW

**Comment by tymofij** [comment]
I would like to see test here, something that disabled post_save signals, then adds few notes, makes sure they are not indexed and issues update_index command to index several of them by time range.

**Comment by jimabramson** [comment]
Agree with @tymofij about the test, and it should cover deletes too.  Also I don't think index_queryset is needed because it does the same as the default according to haystack docs.

**Comment by jimabramson** [comment]
Agree with @tymofij about the test, and it should cover deletes too.  Also I don't think index_queryset is needed because it does the same as the default according to haystack docs.

**Comment by olmar** [comment]
@tymofij  @jimabramson tests added.
Deletion from index is possible only with ``--remove``.  And it seems ``-start/end`` does not play any role when using them with remove flag. See tests for details and this line from django-haystack: https://github.com/toastdriven/django-haystack/blob/master/haystack/management/commands/update_index.py#L244

File: notesapi/v1/tests/test_update_index.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        url = reverse('api:v1:annotations_detail', kwargs={'annotation_id': first_note['id']})
+        response = self.client.delete(url, self.headers)
+
+        # Delete second note.
+        second_start = datetime.datetime.now()
+        url = reverse('api:v1:annotations_detail', kwargs={'annotation_id': second_note['id']})
+        response = self.client.delete(url, self.headers)
+        second_end = datetime.datetime.now() + datetime.timedelta(minutes=10)
+
+        # Try to update when second note was deleted only.
```
Comment (tymofij) [inline notesapi/v1/tests/test_update_index.py]: when only second note was deleted

Comment (olmar) [inline notesapi/v1/tests/test_update_index.py]: > when only second note was deleted
Fixed.

**Comment by tymofij** [comment]
:+1:

**Pull request merged**